### DISCUSSION TITLE
Vulnerability alerts aren't triggered for transistive dependencies in poetry.lock

### DISCUSSION THREAD

**Discussion by freakaton** [renovatebot/renovate#28776] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Self-Hosted Gitlab. Latest version of Renovate(37.333.1)

### Please tell us more about your question or problem

I am configuring Renovate to focus exclusively on security vulnerabilities for a Python project managed with Poetry. 
My current configuration aims to disable all updates except those needed for security patches. However, Renovate appears only to detect and update direct dependencies defined in `pyproject.toml` and misses transitive dependencies. Here is my current Renovate config:
```
{
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": [
      "config:recommended"
   ],
   "packageRules": [
      {
         "enabled": false,
         "groupName": "everything",
         "matchPackagePatterns": [
            "*"
         ]
      }
   ],
   "vulnerabilityAlerts": {
      "enabled": true
   },
   "osvVulnerabilityAlerts": true,
   "dependencyDashboardOSVVulnerabilitySummary": "all",
   "transitiveRemediation": true,
   "assigneesFromCodeOwners": true,
   "prHourlyLimit": 0,
   "enabledManagers": [
      "poetry"
   ]
}
```

And renovate only sees packages, that directly declared in pyproject. (For example I have `black=^22.6.0`, and renovate creates MR with updated black(`^22.6.0` -\> `^24.0.0`) to address PYSEC-2024-48. But I know, that other packages in pyproject have transistive dependencies, that also contain vulnerabilities.

I checked it, by creating `pip freeze > requirements.txt` and enabling the `pip_requirements` manager in config.

And beside black, it also created MR for fastaspi(to address PYSEC-2024-38) as `FastAPI` is a transistive dependency in my project. But this MR is not suitable for me as I use `poetry.lock`, not `requirements.txt` manager.

Questions:
1) How can I configure Renovate to directly detect and update vulnerabilities in transitive dependencies within poetry.lock without resorting to requirements.txt?
2) Is there a way to force Renovate to consider the full dependency graph as resolved in poetry.lock for vulnerability checks?

If you need additional information or examples of pyproject and poetry.lock - I'm ready to share

### Logs (if relevant)

<details><summary>Logs with pip_requirements turned off</summary>

```
DEBUG: fetchVulnerabilities() - osvVulnerabilityAlerts=true (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-fj7x-q9j7-g6q6 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Vulnerability PYSEC-2024-48 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Setting allowed version ==24.3.0 to fix vulnerability GHSA-fj7x-q9j7-g6q6 in black 22.12.0 (repository=renovate/test-repo)
DEBUG: Setting allowed version ==24.3.0 to fix vulnerability PYSEC-2024-48 in black 22.12.0 (repository=renovate/test-repo)
```
</details>

<details><summary>Logs with pip_requirements turned on</summary>

```
DEBUG: fetchVulnerabilities() - osvVulnerabilityAlerts=true (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-fj7x-q9j7-g6q6 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Vulnerability PYSEC-[202](https://********/renovate/renovate-runner/-/jobs/1282641#L202)4-48 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Vulnerability PYSEC-2024-38 affects fastapi 0.103.2 (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-2jv5-9r88-3w3p affects fastapi 0.103.2 (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-fj7x-q9j7-g6q6 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Vulnerability PYSEC-2024-48 affects black 22.12.0 (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-jjg7-2v4v-x38h affects idna 3.6 (repository=renovate/test-repo)
DEBUG: Vulnerability GHSA-2jv5-9r88-3w3p affects starlette 0.27.0 (repository=renovate/test-repo)
```
</details>

**Comment by rarkins** [comment]
1. This is not supported/configurable
2. No

**Reply by freakaton** [reply]
Sounds like an important feature. 
Because right now I can't track vulnerabilities effectively. 
I can help with implementing this

**Reply by stevie-** [reply]
How about the hosted version of the renovate runner?
We see that renovate is not suggesting security updates for the poetry.lock, despite dependabot reports them as an alert.

Do we need to enable `lockFileMaintenance`, so renovate suggests security updates for it?

**Reply by rarkins** [reply]
Yes, enable lockFileMaintenance

**Reply by stevie-** [reply]
We've enable it, but still don't see any security update suggestions in our dashboard.
There is a new entry for lockFileMaintenance
```
 chore(deps): lock file maintenance
```
Also in the dashboard there is in `Detected dependencies` no list of any poetry.lock files mentioned.

**Reply by rarkins** [reply]
Seeing a lock file maintenance entry is the expected result. If you trigger it, it will update all dependencies in the lock file to the latest version.

**Reply by stevie-** [reply]
If I only what to use renovate to detect security related updates in poetry.lock (like a filter).
Is this possible?

**Reply by rarkins** [reply]
No
